app/models/spree/stock/quantifier.rb

```diff
@@ -11,6 +11,8 @@ def initialize(variant)
       end
 
       def total_on_hand
+        return 0 if @variant.deleted?
```

Comment:
maybe useful to have a comment explaining why we need this line?
I imagine something like "if variant is deleted and is still in the cart, the zero total_on_hand will make it out of stock and force it's removal from the cart" ?

Also, and probably more important, I think you need a similar "if super.deleted?" on ScopeVariantToHub#def total_on_hand ?

Reply:
Yeah I was looking at a couple of cases like this. With some minor adjustments we can have overridden stock still function on one enterprise if a variant us deleted on another enterprise. Like if a distributor has 200 stock items for an override, but the variant is soft-deleted by their supplier (maybe as a complete surprise to the distributor) they still have that stock (inventory), and they still need to sell it, right? 

It raises some interesting questions though...

Reply:
Basically if the override explicitly has stock, ScopeVariantToHub#total_on_hand will have valid stock items that could theoretically be used.

Reply:
that makes sense, but I dont think it will work on the shop/cart/checkout right? I think we should keep things very very simple. Maybe we should create an issue to verify that scenario and fix it?

Reply:
I had a quick play with it and I think it will totally work in shop/cart/checkout with some very small adjustments. In another issue though :+1:

I'll add the explanatory comment here for now.

Reply:
nice comment!